services/wallet/activity/activity.go

```diff
@@ -960,3 +976,164 @@ func updateKeypairsAccountsTable(accountsDb *accounts.Database, db *sql.DB) erro
 
 	return nil
 }
+
+func protocolTypeFromDBType(dbType string) (protocolType *ProtocolType) {
+	protocolType = new(ProtocolType)
+	switch common.Type(dbType) {
+	case common.UniswapV2Swap:
+		fallthrough
+	case common.UniswapV3Swap:
+		*protocolType = ProtocolUniswap
+	case common.HopBridgeFrom:
+		fallthrough
+	case common.HopBridgeTo:
+		*protocolType = ProtocolHop
+	default:
+		return nil
+	}
+	return protocolType
+}
+
+func getMultiTxDetails(ctx context.Context, db *sql.DB, id string) (*EntryDetails, error) {
```

Comment:
Does it make sense to have details entry in a separate file? `details.go` maybe?